### ISSUE TITLE
starknet_SimulateTransaction response not conform to rpc0.4.0

### ISSUE THREAD

**Issue by PhilippeR26** [eqlabs/pathfinder#1325]
## Bug description :
During tests in Starknet.js code, I have found this problem :
When I use  `starknet_simulateTransactions` on a contract that **revert,** Pathfinder returns an error object : code + message : `LibraryError: -32603: Execution failed: StarknetErrorCode.TRANSACTION_FAILED`

## To reproduce : 
Typescript (Starknet.js) :
```typescript
const provider = new RpcProvider({ nodeUrl: "http://192.168.1.99:9545/rpc/v0.4" }); //mainnet
const contractAddress = "0x02bD907B978F58ceDf616cFf5CdA213d63Daa3AD28Dd3C1Ea17cA6CF5E1D395F";
const invocation0: Invocations = [
        {
            type: TransactionType.INVOKE,
            contractAddress: contractAddress,
            entrypoint: "test_fail",
            calldata: [100]  
        },
        {
            type: TransactionType.INVOKE,
            contractAddress: contractAddress,
            entrypoint: "test_fail",
            calldata: [200] // this contract reverts when calldata is not equal to 100.
        }
    ]
```
## Expected behavior : 
According to rpc spec 0.4.0 https://github.com/starkware-libs/starknet-specs/blob/0e859ff905795f789f1dfd6f7340cdaf5015acc8/api/starknet_trace_api_openrpc.json#L178-L192 , the behavior should be :
- If the simulation succeed, or if some  transactions revert, the result has to be an array of objects {transaction_trace, fee_estimation}.
  > For transactions that succeed, the transaction_trace is an object of type FUNCTION_INVOCATION.
  > For transactions that revert, the transaction_trace is an object of type {revert_reason: string}.
- Else, the answer is just an error message.

## Configuration : 

Pathfinder 0.7.1, on mainnet.
Starknet.js v5.19.0 connected with "http://192.168.1.99:9545/rpc/v0.4"

**Comment by PhilippeR26** [comment]
I agree that tx_n+1 ... tx_n have non relevant results.
Neverthess, tx_1 ... tx_n have useful results for the user.
I propose to return tx_1 ... tx_n results.

**Comment by kkovaacs** [comment]
This has been fixed in the 0.6.0 revision of the JSON-RPC specification: `starknet_simulateTransactions` now returns traces for reverted transactions -- these are no longer handled as errors.
